File: src/main/java/com/developerphil/adbidea/adb/command/receiver/PrintReceiver.java
package com.developerphil.adbidea.adb.command.receiver;

import com.android.ddmlib.IShellOutputReceiver;
import com.developerphil.adbidea.ui.Utils;
import com.google.common.base.Charsets;

public class PrintReceiver implements IShellOutputReceiver {

    private String mString;

    public final void addOutput(byte[] data, int offset, int length) {
        if (!this.isCancelled()) {
            mString = new String(data, offset, length, Charsets.UTF_8) + "\r\n";
        }
    }

    @Override
    public void flush() {

    }

    public void done() {
    }

    @Override
    public boolean isCancelled() {
        return false;
    }

    @Override
    public String toString() {
        return Utils.isEmpty(mString) ? "" : mString;
    }
}

File: src/main/kotlin/com/developerphil/adbidea/adb/command/receiver/PrintReceiver.kt
package com.developerphil.adbidea.adb.command.receiver

import com.android.ddmlib.IShellOutputReceiver
import com.developerphil.adbidea.ui.Utils.Companion.isEmpty
import com.google.common.base.Charsets

class PrintReceiver : IShellOutputReceiver {
    private var mString: String? = null
    override fun addOutput(data: ByteArray, offset: Int, length: Int) {
        if (!this.isCancelled) {
            mString = String(data, offset, length, Charsets.UTF_8) + "\r\n"
        }
    }

    override fun flush() {}
    override fun isCancelled(): Boolean {
        return false
    }

    override fun toString(): String {
        return if (isEmpty(mString)) "" else mString!!
    }
}